### ISSUE TITLE
no-import-assign should include opt-in restriction for modifying properties on an import

### ISSUE THREAD

**Issue by manufacturedba** [eslint/eslint#13132]
**What rule do you want to change?**

no-import-assign 

**Does this change cause the rule to produce more or fewer warnings?**

More

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option

**Please provide some example code that this change will affect:**

```js
/*eslint no-import-assign: ["error", { "preventMutation": true }]*/
import mod, { named } from "./mod.mjs"
import * as mod_ns from "./mod.mjs"

mod.prop = 1 // ERROR: 'mod' is readonly.
named.prop = 2 // ERROR: 'named' is readonly.
```

**What does the rule currently do for this code?**

Currently, this will produce no error, as there is no reassignment.

**What will the rule do after it's changed?**

Throw errors for any changes to an imported object

**Are you willing to submit a pull request to implement this change?**

Yes

**Comment by mdjermanovic** [comment]
Hi @manufacturedba, thanks for this proposal!

Makes sense to me :+1: 

Should it also report these cases:

```js
mod.prop1.prop2 = 1
named.prop1.prop2 = 2
mod_ns.prop1.prop2 = 3
```

**Comment by kaicataldo** [comment]
This makes sense to me :+1:. Could we call it `props` for parity with [`no-param-reassign`](https://eslint.org/docs/rules/no-param-reassign#options)?

**Comment by manufacturedba** [comment]
> This makes sense to me . Could we call it `props` for parity with [`no-param-reassign`](https://eslint.org/docs/rules/no-param-reassign#options)?

Yup, I think that's a good idea
